.github/workflows/ci.yml

```diff
@@ -344,9 +344,10 @@ jobs:
       - name: Install necessary packages
         # can't use setup-python because that python doesn't seem to work;
         # `python3-dev` (rather than `python:alpine`) for some ctypes reason,
+        # `curl`, `gpg`, `git`, `jq` for codecov-action v4/v5 to work (https://github.com/codecov/codecov-action/issues/1320).
         # `nodejs` for pyright (`node-env` pulls in nodejs but that takes a while and can time out the test).
         # `perl` for a platform independent `sed -i` alternative
-        run: apk update && apk add python3-dev bash nodejs perl
+        run: apk update && apk add python3-dev bash curl gpg git jq nodejs perl
```

Comment:
fyi, `jq` should no longer be needed. I'm working on `gpg` and `git`, though `git` is probably still recommended

Reply:
That's too bad. The best practice is to build and test from sdist, meaning that there would be no Git repo on disk.